pkg/reconciler/revision/resources/pa.go

```diff
@@ -47,13 +47,15 @@ func MakePA(rev *v1.Revision) *autoscalingv1alpha1.PodAutoscaler {
 			Reachability: func() autoscalingv1alpha1.ReachabilityType {
 				// If the Revision has failed to become Ready, then mark the PodAutoscaler as unreachable.
 				if rev.Status.GetCondition(v1.RevisionConditionReady).IsFalse() {
+					if rev.GetRoutingState() == v1.RoutingStatePending || rev.GetRoutingState() == v1.RoutingStateReserve {
+						return autoscalingv1alpha1.ReachabilityUnreachable
+					}
```

Comment:
This change s a noop

https://github.com/knative/serving/blob/ccc40402ae031b293d1a73ebfebe77e0a3b3338d/pkg/apis/serving/v1/revision_helpers.go#L132-L134


ie. if you comment out this change and run the new unit test it doesn't fail